### PR TITLE
Make more accessible the profiler from TruffleRuby on GraalVM

### PR DESCRIPTION
**Pull request by mc2dx**
Related to https://github.com/oracle/truffleruby/issues/2044

Hi there,
During experiments with TruffleRuby and, of course, with Stackprof gem, I think that it would be beneficial if we can show information about the profiler from TruffleRuby/GraalVM. It can help for a newbie and somebody how to discover Ruby/GraalVM environments.
I know that it is no big deal but nice to have a feature.
Please let me know what you think about it.

### CODE REVIEW

File: ext/stackprof/extconf.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 require 'mkmf'
-if have_func('rb_postponed_job_register_one') &&
+if RUBY_ENGINE == 'truffleruby'
+  fail "try truffleruby's profiler \nruby --experimental-options --cpusampler --cpusampler.Mode=roots --cpusampler.SampleInternal FILE.rb"
```
Comment (NickLaMuro) [inline ext/stackprof/extconf.rb:3]: Not being a `TruffleRuby` user myself, I assume that the reasoning for this PR is that `stackprof` is not support on that platform (JVM?).  Maybe include that as part of the string here:

```ruby
fail "stackprof not supported on this platform (#{RUBY_ENGINE})!\n" \
     "for truffleruby, consider trying out truffleruby's profiler:\n\n" \
     "  $ ruby --experimental-options --cpusampler --cpusampler.Mode=roots --cpusampler.SampleInternal FILE.rb"
```

Comment (NickLaMuro) [reply ext/stackprof/extconf.rb:3]: Also, not sure the `extconf.rb` is the best place for this, though I don't have an alternative off hand to suggest.

File: lib/stackprof.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "stackprof/stackprof"
 
 module StackProf
-  VERSION = '0.2.15'
+  VERSION = '0.2.16'
```
Comment (NickLaMuro) [inline lib/stackprof.rb:4]: Generally speaking, I _**think**_ the version bumps have been done outside feature PRs in this repo.  Not positive on that statement, but something to consider.

Comment (mc2dx) [reply lib/stackprof.rb:4]: I "brought" this practice from a different repository, but you are right.

File: ext/stackprof/extconf.rb (hunk 1 above)
Comment (mc2dx) [reply ext/stackprof/extconf.rb:3]: Maybe `mkmf` can be a better place but it is a more complex thing.

**Comment by mc2dx** [comment]
I closed the PR because I am not sure if "discover alternative profilers for TruffleRuby/GraalVM environments" can be achieved by these changes.

**Comment by NickLaMuro** [comment]
@mc2dx Does this maybe make sense to just put in the _Requirements_ section of the `README.md`?

https://github.com/tmm1/stackprof#requirements

Maybe change the requirement to be "C Ruby 2.2+", and add a note after along the lines of:

> For TruffleRuby/GraalVM Rubies, consider using the built in profilers that are available:
> 
> ```console
> $ ruby --experimental-options --cpusampler --cpusampler.Mode=roots --cpusampler.SampleInternal FILE.rb
> ```

**Comment by mc2dx** [comment]
A small change but better than throwing exceptions.

**Pull request closed without merging**